### ISSUE TITLE
Where is index.php and what is its contents?

### ISSUE THREAD

**Issue by derekdiamond** [Algo-Web/POData#212]

**Comment by c-harris** [comment]
I think perhaps you misunderstand the package, this is a translation layer that provides an odata end point. index.php is something that would need to be implemented by you you can find an example to work with a local sql database [here ](https://github.com/Algo-Web/POData-example)

If you give us some more information about your use case, are you using a PHP framework, what data source you are attempting to expose we will be happy to help you.

**Comment by derekdiamond** [comment]
Hi!
Thanks so much for getting back to me!
For now I will be using a MySql data source.
I am doing a disability claims system that I hope to make into a Wordpress plugin. I want to use DevExpress and Syncfusion grid  and other components that all can use OData as a data source. I am really looking for a full CRUD source so I am even willing to update the source to allow full CRUD as needed. I think having an OData Php server/server generator would be an awesome solution because of the large number if WP sites out there and then one has a server accessible to almost any client (java, c#, js and even PHP!)
So basically I want to easily build an OData business layer for my solution. 
If this foes well I intend to use this to move a lot of my spatial solutions to PHP hosting as well.
I hope this gives you an idea of what I am doing. 
I have only recently started learning PHP. My background has been mostly .NET, JS and some Java and C++
Regards
Derek 

Admin Edit to remove extra email gubbins.

**Comment by c-harris** [comment]
@derekdiamond, well the big thing here is where you want the OData layer to interface, through word press or directly to the database. I would say an interfacing through wordpress would probably be the way to go. because then it would be reusable let me give you a basic brake down of how things work and we will see if we can work out how to interface it with wordpress together.

first there is the [BaseService](https://github.com/Algo-Web/POData/blob/master/src/POData/BaseService.php) that class is abstract, it is done that way because the service requires an [IQueryProvider](https://github.com/Algo-Web/POData/blob/master/src/POData/Providers/Query/IQueryProvider.php) and a [IMetadataProvider](https://github.com/Algo-Web/POData/blob/master/src/POData/Providers/Metadata/IMetadataProvider.php)

For simplicity sake we have implomented some basic "simple" classes to avoid the need [SimpleDataService](https://github.com/Algo-Web/POData/blob/master/src/POData/SimpleDataService.php) takes a [IQueryProvider.php](https://github.com/Algo-Web/POData/blob/master/src/POData/Providers/Query/IQueryProvider.php) and a [SimpleMetadataProvider](https://github.com/Algo-Web/POData/blob/master/src/POData/Providers/Metadata/SimpleMetadataProvider.php) you will find that  [SimpleMetadataProvider](https://github.com/Algo-Web/POData/blob/master/src/POData/Providers/Metadata/SimpleMetadataProvider.php) provides the basic application of [IMetadataProvider](https://github.com/Algo-Web/POData/blob/master/src/POData/Providers/Metadata/IMetadataProvider.php)

The reason we have done it this way is a result of the multiple back ends for the data (the query provider allows custom back ends) and need for a metadata provider is that php is an untyped language where as odata requires object typing. in our [POData-Laravel](https://github.com/Algo-Web/POData-Laravel) project we get the type information directly from the database.

I am happy to work with you todo an official plugin to expose wordpress data directly over odata but you would probably need to direct the project as out dev team don't often use wordpress. Let me know what you think and ill get a repo create?

**Comment by derekdiamond** [comment]
Ok wow that sounds great.

 

Yes I am totally keen. I’ll see how Wordpress fits in on the data layer.

 

Yes lets get a repo going. If we get a WP plugin going we can monetize it together as well, lets see.

 

Regards

 

Derek

 Admin Edit to remove extra email gubbins.

**Comment by c-harris** [comment]
@derekdiamond do you have a better way to communicate then github issues? (Discord,Skype,Etc)

**Comment by derekdiamond** [comment]
Yes, you can get me on Skype DerekcDiamond.

**Comment by CyberiaResurrection** [comment]
Closing due to lack of activity
